ethclient/gethclient/gethclient.go

```diff
@@ -0,0 +1,155 @@
+package gethclient
+
+import (
+	"context"
+	"math/big"
+	"runtime"
+	"runtime/debug"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// Client is a wrapper around the rpc.Client that
+// implements geth specific functionalities.
+// Please use ethclient.Client instead if you want to
+// access the standardized Etheeum RPC functionality.
+type Client struct {
+	c *rpc.Client
+}
+
+// Dial connects a client to the given URL.
+func Dial(rawurl string) (*Client, error) {
```

Comment:
We should avoid adding the Dial* functions for gethclient. I added them in ethclient for convenience, but it kind of made people prefer them instead of rpc.Dial, which has more options. 

Since gethclient is limited to very specific functionality, users will pretty much always need to rpc.Dial and then call NewClient to make the wrapper.